test-utils/junit5-extension/src/main/java/org/camunda/bpm/engine/test/junit5/ProcessEngineExtension.java

```diff
@@ -88,19 +106,138 @@
+    filterService = processEngine.getFilterService();
+    externalTaskService = processEngine.getExternalTaskService();
+    decisionService = processEngine.getDecisionService();
+  }
+
+  protected void clearServiceReferences() {
+    processEngineConfiguration = null;
+    repositoryService = null;
+    runtimeService = null;
+    taskService = null;
+    formService = null;
+    historyService = null;
+    identityService = null;
+    managementService = null;
+    authorizationService = null;
+    caseService = null;
+    filterService = null;
+    externalTaskService = null;
+    decisionService = null;
+  }
+
+  // TEST EXECUTION
+
+  @Override
+  public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+    return parameterContext.getParameter().getType().equals(ProcessEngine.class);
+  }
+
+  @Override
+  public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+    if (ProcessEngine.class.equals(parameterContext.getParameter().getType())) {
+      LOG.debug("resolve the processEngine as parameter");
+      return getProcessEngine();
+    } else {
+      return null;
+    }
+  }
+
+  @Override
+  public void beforeTestExecution(ExtensionContext context) {
+    LOG.debug("beforeTestExecution: {}", context.getDisplayName());
+
+    initializeServices();
+
+    final Method testMethod = context.getTestMethod().orElseThrow(illegalStateException("testMethod not set"));
+    final Class<?> testClass = context.getTestClass().orElseThrow(illegalStateException("testClass not set"));
+
+    deploymentId = TestHelper.annotationDeploymentSetUp(processEngine, testClass, testMethod.getName(), null, testMethod.getParameterTypes());
+    boolean hasRequiredHistoryLevel = TestHelper.annotationRequiredHistoryLevelCheck(processEngine, testClass, testMethod.getName(), testMethod.getParameterTypes());
+    boolean hasRequiredDatabase = TestHelper.annotationRequiredDatabaseCheck(processEngine, testClass, testMethod.getName(), testMethod.getParameterTypes());
+
+    Assumptions.assumeTrue(hasRequiredHistoryLevel, "ignored because the current history level is too low");
+    Assumptions.assumeTrue(hasRequiredDatabase, "ignored because the database doesn't match the required ones");
+  }
+
+  @Override
+  public void afterTestExecution(ExtensionContext context) {
+    final String testMethod = context.getTestMethod().orElseThrow(illegalStateException("testMethod not set")).getName();
+    final Class<?> testClass = context.getTestClass().orElseThrow(illegalStateException("testClass not set"));
+
+   TestHelper.annotationDeploymentTearDown(processEngine, deploymentId, testClass, testMethod);
+   deploymentId = null;
+   for (String additionalDeployment : additionalDeployments) {
+     TestHelper.deleteDeployment(processEngine, additionalDeployment);
+   }
```

Comment:
```suggestion
   }
   additionalDeployments.clear();
```
Why should we keep the deployments?

Reply:
Nice catch!